lib/charms/layer/jenkins/api.py

```diff
@@ -105,10 +108,22 @@ def get_plugin_version(self, plugin):
             return False
         return version
 
-    def configure_proxy(self, hostname=None, port=None, username=None, password=None):
+    def configure_proxy(
+        self, hostname=None, port=None, username=None, password=None, no_proxy_hosts=None
+    ):
         """Configure (or disable) a system proxy."""
         client = self._make_client()
-        if username and password and hostname and port:
+        if username and password and hostname and port and no_proxy_hosts:
```

Comment:
What happens if you want to use a proxy without auth but still want to set a no-proxy value?

Reply:
Fixed to include the case, thank you!